pallets/gear/src/lib.rs

```diff
@@ -1842,8 +1850,12 @@ pub mod pallet {
         type Gas = GasBalanceOf<T>;
 
         fn run_queue(initial_gas: Self::Gas) -> Self::Gas {
-            // Setting adjusted initial gas allowance
-            GasAllowanceOf::<T>::put(initial_gas);
+            // Take note of how much gas we will need for builtin router creation.
+            let gas_for_builtin_router = T::BuiltinRouter::provision_cost();
+
+            // Setting adjusted initial gas allowance taking into account builtin router creation
+            // cost that will be incurred inside the `process_queue()` function.
+            GasAllowanceOf::<T>::put(initial_gas.saturating_sub(gas_for_builtin_router));
```

Comment:
found the place of this charge, I'd really provide() here in place to keep it together for the eyes

Reply:
**Update**: now these two actions (creating a `BuiltinDispatcher` and calculating the cost of this operation) are tied together.